src/components/vm/vmRenameDialog.jsx

```diff
@@ -23,47 +23,57 @@ import { Button } from "@patternfly/react-core/dist/esm/components/Button";
 import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
 import { Modal } from "@patternfly/react-core/dist/esm/components/Modal";
 import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { TextArea } from "@patternfly/react-core/dist/esm/components/TextArea";
 
 import { FormHelper } from 'cockpit-components-form-helper.jsx';
 import { ModalError } from 'cockpit-components-inline-notification.jsx';
 import { useDialogs } from 'dialogs.jsx';
 
 import { isObjectEmpty } from '../../helpers.js';
-import { domainRename } from '../../libvirtApi/domain.js';
+import { domainRename, domainSetMetadata } from '../../libvirtApi/domain.js';
 
 const _ = cockpit.gettext;
 
-export const RenameDialog = ({ vmName, vmId, connectionName }) => {
+export const RenameDialog = ({ vm }) => {
```

Comment:
Note: This is convenient, but `vm` objects have a ton of properties and change all the time (like runtime metrics). This will re-render the dialog a lot.

Reply:
Does it? Components get rendered when the parent is rendered or when their state changes, not when a field of one of their props changes.

Reply:
Any change to the redux store renders all of cockpit-machines, which is maybe not great, but this change here doesn't make that worse, I'd say.

Reply:
> Components get rendered when the parent is rendered or when their state changes

.. or when their properties change (which is one of the two main functions of React).

> Any change to the redux store renders all of cockpit-machines

Right, fair enough.

Reply:
> > Components get rendered when the parent is rendered or when their state changes
> 
> .. or when their properties change (which is one of the two main functions of React).

Hmm. Let me take this as an opportunity to figure this out for myself...

My mental model is this: Properties change as a result of rendering. That's the only way they can change. Property changes do not cause rendering. All rendering is triggered by a change in _state_ in some component, and then the whole tree starting at that component is re-rendered.

Components in the tree can optionally stop their re-rendering with `React.memo` (or by deriving from React.PureComponent), but that is not the default and we don't use this here.

And because c-machines uses Redux, comparing objects with Object.is is correct and wont miss changes. So I guess all our components could use React.memo by default... Is there a way to switch that on? Or maybe it is already?

In any case, I don't see how changing from name/connection to store object would change how often the dialog is rendered.

https://www.debugbear.com/blog/react-rerenders

(There might be other rules where React itself by default decides to stop rendering at a node in the subtree, but I haven't found anything concrete about this, unfortunately. React is surprisingly wishy-washy about what the rules for rendering actually are... But it doesn't compare property values, I am pretty sure. We rely on that in many many places. There is some magic with updating the state... this.setState({}) used to reliably trigger a render, but it's harder with the `useState` hook...)